src/core/api/netdata_api.cpp

```diff
@@ -106,3 +108,45 @@ uint8_t otNetDataGetStableVersion(otInstance *aInstance)
 
     return instance.Get<Mle::MleRouter>().GetLeaderData().GetStableDataVersion();
 }
+
+static otError SteeringDataCheck(Instance &aInstance, FilterIndexes &aFilterIndexes)
```

Comment:
The code looks fine. 

One suggest to add such methods in `NetworkData::Leader>` (or maybe some other core class) and call them from here?

I think would be good to keep the `{module}_api.cpp` source files simple and have them just call some internal method (on related object/class) and have the main implementation in the core modules.

Reply:
updated.